core/primitives/src/state_part.rs

```diff
@@ -10,3 +23,177 @@ impl PartId {
         PartId { idx: part_id, total: num_parts }
     }
 }
+
+/// Serialized version of `PartialState`.
+#[derive(Debug, Clone, PartialEq, Eq, BorshSerialize, BorshDeserialize, ProtocolSchema)]
+pub struct StatePartV0(pub(crate) Vec<u8>);
+
+/// Similar to `StatePartV0`, but uses zstd compression.
+#[derive(Clone, PartialEq, Eq, BorshSerialize, BorshDeserialize, ProtocolSchema)]
+pub struct StatePartV1 {
+    bytes_compressed: Vec<u8>,
+}
+
+#[derive(Clone, PartialEq, Eq, BorshSerialize, BorshDeserialize, ProtocolSchema, AsRefStr)]
+#[borsh(use_discriminant = true)]
+#[repr(u8)]
+pub enum StatePart {
+    /// Borsh-serialized trie nodes
+    V0(StatePartV0) = 0,
+    /// zstd-compressed borsh-serialized trie nodes
+    V1(StatePartV1) = 1,
+}
+
+impl fmt::Debug for StatePart {
```

Comment:
I think this was missed the last time, we can get rid of the custom Debug function and derive it instead. We can also get rid of `AsRefStr`